rfcs/0001-tokens-naming-convention.md

```diff
@@ -0,0 +1,495 @@
+
+A `Modifier` can be further split into following sub-categories based on the granularity of informatin we want to store in our token and follows the hierarchy as:
+1. **variant** - `primary`, `secondary` etc.
+2. **state** - `hover`, `click`, `active` etc.
+3. **scale** - `100`, `200`, `1`, `2`, `s`, `m`, `l` etc.
+4. **mode** - `dark`, `light` etc.
+
+You can use multiple `modifier` in the same token to add more granularity and you can even skip any of the sub-categories but the hierarchical levels needs to be intact as mentioned above. For example you can construct a token by pairing the sub-categories of `modifier` as `variant.state.scale.mode` or `variant.scale.mode` or `variant.state.mode` or `variant.mode` and so on.
+
+You can basically use any/all of the above categories and their sub-categories and pair them up to form a very specific self explanatory token. Below are some examples:
+```js
+Button.font.size.primary.disabled.m
+Button.color.text.primary.hover.dark
+Button.color.text.primary
+theme.font.family
+theme.font.size
+theme.font.weight
+theme.color.chromatic.azure.100
+theme.space.s
+```
+
+With this system If I take the example mentioned earlier about the usage this is how it will look
+```js
+const IconTokens = {
+  color: {
+    text: {
+      primary: {
+        value: theme.color.light[900],
+        disabled: {
+          value: theme.color.light[950],
+        },
+      },
+      secondary: {
+        value: theme.color.shade[800],
+        disabled: {
+          value: theme.color.shade[940],
+        },
+      },
+      tertiary: {
+        value: theme.color.shade[800],
+        disabled: {
+          value: theme.color.shade[940],
+        },
+      },
+      value: theme.color.light[900],
+      disabled: {
+        value: theme.color.light[950],
+      },
+    },
+  },
+};
+fontColor({ variant, variantColor, disabled }) {
+  switch (variant) {
+    case 'primary':
+      if (disabled) {
+        return IconTokens.color.text.primary.disabled.value;
+      }
+      return IconTokens.color.text.primary.value;
+    case 'secondary':
+      if (disabled) {
+        return IconTokens.color.text.secondary.disabled.value;
+      }
+      return IconTokens.color.text.secondary.value;
+    case 'tertiary':
+      if (disabled) {
+        return IconTokens.color.text.tertiary.disabled.value;
+      }
+      return IconTokens.color.text.tertiary.value;
+    default:
+      if (disabled) {
+        return IconTokens.color.text.disabled.value;
+      }
+      return IconTokens.color.text.value;
+  }
+}
+// and then we use it like this
+<Icon
+  name={icon}
+  size={styles.iconSize({ size, children })}
+  fill={styles.fontColor({ variant, variantColor, disabled })}
+  testID="Button-left-icon"
+/>
+```
+
+>P.S Some work still needs to be done. I need to figure out if we can completely get rid of styles.fontColor function
```

Comment:
I think there still might be certain edge if/else conditions based on prop values and their derivations. Let's keep this function for now. After building some components, if we feel this is getting redundant we can remove it.